core/src/test/java/knaccc/monero/address/WalletAddressTest.java

```diff
@@ -46,5 +47,28 @@ public void testWalletAddress() throws WalletAddress.InvalidWalletAddressExcepti
         assertEquals(walletAddress.getSubaddressBase58(privateViewKeyHex, 0, 1), addr01);
         assertEquals(walletAddress.getSubaddressBase58(privateViewKeyHex, 1, 0), addr10);
         assertEquals(walletAddress.getSubaddressBase58(privateViewKeyHex, 1, 1), addr11);
+
+        assertEquals(walletAddress.checkPrivateViewKey(privateViewKeyHex), true);
+        assertEquals(WalletAddress.arePubPrivKeysRelated(publicViewKeyHex, privateViewKeyHex), true);
+        assertEquals(WalletAddress.arePubPrivKeysRelated(privateViewKeyHex, privateViewKeyHex), false);
+
+        assertEquals(WalletAddress.arePubPrivKeysRelated(
+                "bdc158199c8933353627d54edb4bbae547dbbde3130860d7940313210edca0a6",
+                "a82a9017a1d259c71f5392ad9091b743b86dac7a21f5e402ea0a55e5c8a6750f"),
+                true);
+
+        assertEquals(WalletAddress.arePubPrivKeysRelated(
+                "d17698d07fe9edbc41552299b90a93de73bb1bd4b94b8083af0bbe3a1931e2ec",
+                "dae1bceeb2563b8c376f8e0456e5fe7aa3d6291b38ace18c6ad5647424a3b104"),
+                true);
+
+        assertEquals(WalletAddress.arePubPrivKeysRelated(
+                "0000111122223333444455556666777788889999AAAABBBBCCCCDDDDEEEEFFFF",
+                "0000111122223333444455556666777788889999AAAABBBBCCCCDDDDEEEEFFFF"),
+                false);
+
+        String nonReducedPrivateKey = "42594aba0e809490dec97b2dfaf64f7ae5bef1c2d19af636eb84544773df5b5f";
```

Comment:
Can you use "680bceef3ca8b2ca1a9a29283c184f6f590a9bd2881825f0542ad99cdaba091a" instead, this scalar  reduces to the private view key of the wallet being tested against.